### ISSUE TITLE
getopts: detect when OPTARG is used without colon in optstring

### ISSUE THREAD

**Issue by wwuck** [koalaman/shellcheck#1989]
#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

usage() {
        echo "$0 [-a text] [-b]"
        exit 1
}

while getopts "ab" opt; do
        case "$opt" in
                a) foo=$OPTARG ;;
                b) alias bar=true ;;
                *) usage ;;
        esac
 done

echo "$foo"

```

#### Here's what shellcheck currently says:

`No issues detected!`

#### Here's what I wanted or expected to see:

Something similar to `OPTARG used in option case a but no colon specified after a in optstring`.

Shellcheck should warn that `foo` is being assigned to $OPTARG when there is no colon after `a` in the optstring.  This could lead to the user expecting `foo` to get a value but no value being passed to the script command line arguments.

**Comment by qneill-sifive** [comment]
Probably should detect the opposite too: when a colon is present but $OPTARG is NOT used.
